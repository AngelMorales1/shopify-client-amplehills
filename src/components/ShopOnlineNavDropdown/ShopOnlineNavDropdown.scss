@import 'settings', 'typography', 'src/styles/utils';

.ShopOnlineNavDropdown {
  @extend .transition-dropdown-slide-down;
  display: none;

  &--active {
    display: inherit;
  }

  &__nav-container {
    top: $header-height-desktop;
    height: spacing(70);
  }

  &__nav-container--with-alert {
    top: calc(#{$header-height-desktop} + #{$alert-height-desktop});
  }

  &__card {
    width: spacing(42);
    height: spacing(55);
    border-radius: radius('xs');
    overflow: hidden;
  }

  &__image--container {
    height: 70%;
  }

  &__image {
    object-fit: contain;
  }

  &__card-text-container {
    font-size: $font-2;
    line-height: $font-4;
    height: 20%;
  }

  &__merch-container {
    height: spacing(55);
  }
}
