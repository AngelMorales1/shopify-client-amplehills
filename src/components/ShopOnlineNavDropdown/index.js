import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import get from 'utils/get';

import styles from './ShopOnlineNavDropdown.scss';
import { Image, Button } from 'components/base';

const ShopOnlineNavDropdown = ({
  productLanding,
  alertIsActive,
  shopOnlineDropdownIsOpen,
  openShopOnline,
  closeShopOnline
}) => {
  const iceCreamProducts = get(productLanding, 'iceCreamProducts', []);
  const merchProducts = get(productLanding, 'merchandiseProducts', []);

  return (
    <div
      onMouseEnter={openShopOnline}
      onMouseLeave={closeShopOnline}
      className={cx(
        styles['ShopOnlineNavDropdown'],
        {
          [styles['ShopOnlineNavDropdown--active']]: shopOnlineDropdownIsOpen
        },
        'relative transition-dropdown-slide-down'
      )}
    >
      <div
        className={cx(
          styles['ShopOnlineNavDropdown__nav-container'],
          {
            [styles[
              'ShopOnlineNavDropdown__nav-container--with-alert'
            ]]: alertIsActive
          },
          'absolute z-nav'
        )}
      >
        <div
          className={cx(
            'w100 pt3 px4 bg-peach fixed drip flex flex-row items-center'
          )}
        >
          <div className="col-6 flex flex-row justify-center mx-auto">
            {iceCreamProducts.map(product => {
              const fields = get(product, 'fields', {});
              const image = get(fields, 'dropdownNavImage.fields.file.url', '');
              const title = get(fields, 'productTitle', '');
              const handle = get(fields, 'productHandle', '');

              return (
                <Button
                  onClick={closeShopOnline}
                  key={handle}
                  variant="style-none"
                  to={`/products/${handle}`}
                >
                  <div
                    className={cx(
                      styles['ShopOnlineNavDropdown__card'],
                      'bg-white mx2 p3 flex flex-column items-center justify-between'
                    )}
                  >
                    <div
                      className={cx(
                        styles['ShopOnlineNavDropdown__image--container'],
                        'w100'
                      )}
                    >
                      <Image
                        className={cx(
                          styles['ShopOnlineNavDropdown__image'],
                          'wh100'
                        )}
                        src={image}
                      />
                    </div>
                    <div
                      className={cx(
                        styles['ShopOnlineNavDropdown__card-text-container'],
                        'flex flex-row items-center'
                      )}
                    >
                      <p className="white-space-normal center text-heavy-gray">
                        {title}
                      </p>
                    </div>
                  </div>
                </Button>
              );
            })}
          </div>
          <div
            className={cx(
              styles['ShopOnlineNavDropdown__merch-container'],
              'col-2 absolute r0'
            )}
          >
            <p className="text-white bold">Merch</p>
            {merchProducts.map(merch => {
              const fields = get(merch, 'fields', {});
              const title = get(fields, 'title', '');
              const handle = get(fields, 'handle', '');
              return (
                <Button
                  onClick={closeShopOnline}
                  key={handle}
                  variant="style-none"
                  to={`/merchandise/${handle}`}
                >
                  <p className="text-white light mt1">{title}</p>
                </Button>
              );
            })}
          </div>
        </div>
      </div>
    </div>
  );
};

export default ShopOnlineNavDropdown;

ShopOnlineNavDropdown.propTypes = {
  productLanding: PropTypes.object,
  alertIsActive: PropTypes.bool,
  shopOnlineDropdownIsOpen: PropTypes.bool,
  onMouseEnter: PropTypes.func,
  onMouseLeave: PropTypes.func
};

ShopOnlineNavDropdown.defaultProps = {
  productLanding: {},
  alertIsActive: false,
  shopOnlineDropdownIsOpen: false,
  onMouseEnter: () => {},
  onMouseLeave: () => {}
};
