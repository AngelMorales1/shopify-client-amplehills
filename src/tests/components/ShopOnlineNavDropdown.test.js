import React from 'react';
import { shallow } from 'enzyme';

import ShopOnlineNavDropdown from 'components/ShopOnlineNavDropdown';

it('renders without data', () => {
  const component = shallow(<ShopOnlineNavDropdown />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <ShopOnlineNavDropdown
      productLanding={{
        description: 'foo',
        hideWholesaleBlock: false,
        iceCreamDescription: '',
        iceCreamProducts: [],
        iceCreamTitle: 'foo',
        merchandiseDescription: 'foo',
        merchandiseProducts: [],
        merchandiseTitle: '',
        products: [],
        subNavigation: true,
        title: 'foo',
        wholesaleDescription: 'foo',
        wholesaleImage: {
          sys: { id: 'foo' },
          fields: { file: { url: 'foo' } }
        },
        wholesaleTitle: 'foo'
      }}
      alertIsActive={false}
      shopOnlineDropdownIsOpen={false}
      onMouseEnter={() => console.log('foo')}
      onMouseLeave={() => console.log('foo')}
    />
  );

  expect(component).toMatchSnapshot();
});
