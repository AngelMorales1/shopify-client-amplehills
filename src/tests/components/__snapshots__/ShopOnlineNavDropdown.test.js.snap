// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShopOnlineNavDropdown
    alertIsActive={false}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    productLanding={
        Object {
            "description": "foo",
            "hideWholesaleBlock": false,
            "iceCreamDescription": "",
            "iceCreamProducts": Array [],
            "iceCreamTitle": "foo",
            "merchandiseDescription": "foo",
            "merchandiseProducts": Array [],
            "merchandiseTitle": "",
            "products": Array [],
            "subNavigation": true,
            "title": "foo",
            "wholesaleDescription": "foo",
            "wholesaleImage": Object {
              "fields": Object {
                "file": Object {
                  "url": "foo",
                },
              },
              "sys": Object {
                "id": "foo",
              },
            },
            "wholesaleTitle": "foo",
          }
    }
    shopOnlineDropdownIsOpen={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="absolute z-nav"
>
        <div
                className="w100 pt3 px4 bg-peach fixed drip flex flex-row items-center"
        >
                <div
                        className="col-6 flex flex-row justify-center mx-auto"
                >
                        
                </div>
                <div
                        className="col-2 absolute r0"
                >
                        <p
                                className="text-white bold"
                        >
                                Merch
                        </p>
                </div>
        </div>
</div>,
      "className": "relative z-nav",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w100 pt3 px4 bg-peach fixed drip flex flex-row items-center"
>
          <div
                    className="col-6 flex flex-row justify-center mx-auto"
          >
                    
          </div>
          <div
                    className="col-2 absolute r0"
          >
                    <p
                              className="text-white bold"
                    >
                              Merch
                    </p>
          </div>
</div>,
        "className": "absolute z-nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-6 flex flex-row justify-center mx-auto"
>
              
</div>,
            <div
              className="col-2 absolute r0"
>
              <p
                            className="text-white bold"
              >
                            Merch
              </p>
</div>,
          ],
          "className": "w100 pt3 px4 bg-peach fixed drip flex flex-row items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "col-6 flex flex-row justify-center mx-auto",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="text-white bold"
>
                  Merch
</p>,
                Array [],
              ],
              "className": "col-2 absolute r0",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Merch",
                  "className": "text-white bold",
                },
                "ref": null,
                "rendered": "Merch",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="absolute z-nav"
>
          <div
                    className="w100 pt3 px4 bg-peach fixed drip flex flex-row items-center"
          >
                    <div
                              className="col-6 flex flex-row justify-center mx-auto"
                    >
                              
                    </div>
                    <div
                              className="col-2 absolute r0"
                    >
                              <p
                                        className="text-white bold"
                              >
                                        Merch
                              </p>
                    </div>
          </div>
</div>,
        "className": "relative z-nav",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="w100 pt3 px4 bg-peach fixed drip flex flex-row items-center"
>
            <div
                        className="col-6 flex flex-row justify-center mx-auto"
            >
                        
            </div>
            <div
                        className="col-2 absolute r0"
            >
                        <p
                                    className="text-white bold"
                        >
                                    Merch
                        </p>
            </div>
</div>,
          "className": "absolute z-nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-6 flex flex-row justify-center mx-auto"
>
                
</div>,
              <div
                className="col-2 absolute r0"
>
                <p
                                className="text-white bold"
                >
                                Merch
                </p>
</div>,
            ],
            "className": "w100 pt3 px4 bg-peach fixed drip flex flex-row items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "col-6 flex flex-row justify-center mx-auto",
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="text-white bold"
>
                    Merch
</p>,
                  Array [],
                ],
                "className": "col-2 absolute r0",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Merch",
                    "className": "text-white bold",
                  },
                  "ref": null,
                  "rendered": "Merch",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShopOnlineNavDropdown
    alertIsActive={false}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    productLanding={Object {}}
    shopOnlineDropdownIsOpen={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="absolute z-nav"
>
        <div
                className="w100 pt3 px4 bg-peach fixed drip flex flex-row items-center"
        >
                <div
                        className="col-6 flex flex-row justify-center mx-auto"
                >
                        
                </div>
                <div
                        className="col-2 absolute r0"
                >
                        <p
                                className="text-white bold"
                        >
                                Merch
                        </p>
                </div>
        </div>
</div>,
      "className": "relative z-nav",
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="w100 pt3 px4 bg-peach fixed drip flex flex-row items-center"
>
          <div
                    className="col-6 flex flex-row justify-center mx-auto"
          >
                    
          </div>
          <div
                    className="col-2 absolute r0"
          >
                    <p
                              className="text-white bold"
                    >
                              Merch
                    </p>
          </div>
</div>,
        "className": "absolute z-nav",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-6 flex flex-row justify-center mx-auto"
>
              
</div>,
            <div
              className="col-2 absolute r0"
>
              <p
                            className="text-white bold"
              >
                            Merch
              </p>
</div>,
          ],
          "className": "w100 pt3 px4 bg-peach fixed drip flex flex-row items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "col-6 flex flex-row justify-center mx-auto",
            },
            "ref": null,
            "rendered": Array [],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="text-white bold"
>
                  Merch
</p>,
                Array [],
              ],
              "className": "col-2 absolute r0",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Merch",
                  "className": "text-white bold",
                },
                "ref": null,
                "rendered": "Merch",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="absolute z-nav"
>
          <div
                    className="w100 pt3 px4 bg-peach fixed drip flex flex-row items-center"
          >
                    <div
                              className="col-6 flex flex-row justify-center mx-auto"
                    >
                              
                    </div>
                    <div
                              className="col-2 absolute r0"
                    >
                              <p
                                        className="text-white bold"
                              >
                                        Merch
                              </p>
                    </div>
          </div>
</div>,
        "className": "relative z-nav",
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="w100 pt3 px4 bg-peach fixed drip flex flex-row items-center"
>
            <div
                        className="col-6 flex flex-row justify-center mx-auto"
            >
                        
            </div>
            <div
                        className="col-2 absolute r0"
            >
                        <p
                                    className="text-white bold"
                        >
                                    Merch
                        </p>
            </div>
</div>,
          "className": "absolute z-nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-6 flex flex-row justify-center mx-auto"
>
                
</div>,
              <div
                className="col-2 absolute r0"
>
                <p
                                className="text-white bold"
                >
                                Merch
                </p>
</div>,
            ],
            "className": "w100 pt3 px4 bg-peach fixed drip flex flex-row items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "col-6 flex flex-row justify-center mx-auto",
              },
              "ref": null,
              "rendered": Array [],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="text-white bold"
>
                    Merch
</p>,
                  Array [],
                ],
                "className": "col-2 absolute r0",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Merch",
                    "className": "text-white bold",
                  },
                  "ref": null,
                  "rendered": "Merch",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
